pkg/pq/priority_queue_test.go

```diff
@@ -75,6 +78,40 @@ func TestPQ(t *testing.T) {
 			assert.Equal(t, NewTestValue(i), pq.Peek())
 			assert.Equal(t, NewTestValue(i), pq.Pop())
 		}
+		assert.Equal(t, 0, pq.Len())
+
+		// Release
+		for i := 1; i <= 12; i++ {
+			pq.Push(NewTestValue(i))
+		}
+
+		for i := 1; i <= 12; i++ {
+			pq.Release(NewTestValue(i))
+			assert.False(t, exist(i, pq.Values()))
+		}
+		assert.Equal(t, 0, pq.Len())
+
+		for i := 1; i <= 12; i++ {
+			pq.Push(NewTestValue(i))
+		}
+
+		for i := 12; i >= 1; i-- {
+			pq.Release(NewTestValue(i))
+			assert.False(t, exist(i, pq.Values()))
+		}
+		assert.Equal(t, 0, pq.Len())
+
+		for i := 1; i <= 12; i++ {
+			pq.Push(NewTestValue(i))
+		}
+
```

Comment:
How about separating some of the test cases in this test suite? If we separate it, in my view, I think the test code will be easier to read :) 

For example, 
t.Run("priority queue peek test")
t.Run("priority queue pop test")
t.Run("priority queue release test")

Reply:
cool It would be better